joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -84,31 +85,55 @@ JointTrajectoryController::update()
     return CONTROLLER_INTERFACE_RET_SUCCESS;
   }
 
-  // when no traj msg has been received yet
-  if (!traj_point_active_ptr_ || (*traj_point_active_ptr_)->is_empty()) {
-    return CONTROLLER_INTERFACE_RET_SUCCESS;
-  }
+  // when we receive a traj msg
+  if (traj_point_active_ptr_ && (*traj_point_active_ptr_)->is_empty() == false) {
+    // find next new point for current timestamp
+    auto traj_point_ptr = (*traj_point_active_ptr_)->sample(lifecycle_node_->now());
+    // find next new point for current timestamp
+    // set cmd only if a point is found
+    if (traj_point_ptr == (*traj_point_active_ptr_)->end()) {
+      return CONTROLLER_INTERFACE_RET_SUCCESS;
+    }
 
-  // find next new point for current timestamp
-  auto traj_point_ptr = (*traj_point_active_ptr_)->sample(rclcpp::Clock().now());
-  // find next new point for current timestamp
-  // set cmd only if a point is found
-  if (traj_point_ptr == (*traj_point_active_ptr_)->end()) {
-    return CONTROLLER_INTERFACE_RET_SUCCESS;
-  }
+    // check if new point ptr points to the same as previous point
+    if (prev_traj_point_ptr_ == traj_point_ptr) {
+      return CONTROLLER_INTERFACE_RET_SUCCESS;
+    }
 
-  // check if new point ptr points to the same as previous point
-  if (prev_traj_point_ptr_ == traj_point_ptr) {
-    return CONTROLLER_INTERFACE_RET_SUCCESS;
-  }
+    size_t joint_num = registered_joint_cmd_handles_.size();
+    for (size_t index = 0; index < joint_num; ++index) {
+      registered_joint_cmd_handles_[index]->set_cmd(traj_point_ptr->positions[index]);
+    }
 
-  size_t joint_num = registered_joint_cmd_handles_.size();
-  for (size_t index = 0; index < joint_num; ++index) {
-    registered_joint_cmd_handles_[index]->set_cmd(traj_point_ptr->positions[index]);
+    prev_traj_point_ptr_ = traj_point_ptr;
+    set_op_mode(hardware_interface::OperationMode::ACTIVE);
   }
 
-  prev_traj_point_ptr_ = traj_point_ptr;
-  set_op_mode(hardware_interface::OperationMode::ACTIVE);
+  // publish state
+  {
```

Comment:
This block could be extracted into a separate function

Reply:
I'm an advocate for not exposing extra functions to a higher/wider level namespace unless needed. That being said, I'll do so if many more insists on it.

Reply:
There's nothing wrong with creating a private function within the class, that wouldn't expose it, only keep it a little more tidy.

Reply:
The thing is I have to scroll/jump to the function via editor, which is likely a number of lines away (in larger codebases it gets annoying as the number of these 'only used in one place functions' increases), thus breaking concentration quite abit.

But anyway in the interests of keeping things moving along, fixed in [6e00b03](https://github.com/ros-controls/ros2_controllers/pull/25/commits/6e00b03b0e84207812733fcbdfee1f436c95cd60)